plugins/out_s3/s3.c

```diff
@@ -2209,8 +2288,17 @@ static void cb_s3_flush(struct flb_event_chunk *event_chunk,
     }
 
     /* Buffer current chunk in filesystem and wait for next chunk from engine */
-    ret = buffer_chunk(ctx, upload_file, chunk, chunk_size,
-                       event_chunk->tag, flb_sds_len(event_chunk->tag));
+    if (!upload_file) {
+        ret = buffer_chunk(ctx, upload_file, chunk, chunk_size,
+                           event_chunk->tag, flb_sds_len(event_chunk->tag),
+                           &tms.tm.tv_sec);
+    }
+    else {
+        ret = buffer_chunk(ctx, upload_file, chunk, chunk_size,
+                           event_chunk->tag, flb_sds_len(event_chunk->tag),
+                           upload_file->first_log_time);
+    }
+
```

Comment:
Please move this conditional logic into the buffer_chunk function. It's confusing what the timep variable means.

You should always do:
```
    ret = buffer_chunk(ctx, upload_file, chunk, chunk_size,
                       event_chunk->tag, flb_sds_len(event_chunk->tag),
                       time_t chunk_first_log_time
);
```